webapp/tests/karma/unit/services/telemetry.js

```diff
@@ -0,0 +1,215 @@
+describe('Telemetry service', () => {
+  'use strict';
+
+  const NOW = new Date(2018, 10, 10, 12, 33).getTime();
+
+  let clock,
+    service,
+    $window,
+    DB,
+    storageGetItem,
+    storageSetItem,
+    storageRemoveItem,
+    pouchDb;
+
+  beforeEach(() => {
+    module('inboxApp');
+    clock = sinon.useFakeTimers(NOW);
+
+    storageGetItem = sinon.stub();
+    storageSetItem = sinon.stub();
+    storageRemoveItem = sinon.stub();
+
+    pouchDb = {};
+
+    $window = {
+      localStorage: {
+        getItem: storageGetItem,
+        setItem: storageSetItem,
+        removeItem: storageRemoveItem,
+      },
+      navigator: {},
+      PouchDB: () => pouchDb,
+    };
+
+    DB = {
+      info: sinon.stub(),
+      put: sinon.stub(),
+    };
+
+    module($provide => {
+      $provide.value('$q', Q);
+      $provide.value('$window', $window);
+      $provide.factory('DB', KarmaUtils.mockDB(DB));
+      $provide.value('Session', {
+        userCtx: () => {
+          return { name: 'greg' };
+        },
+      });
+    });
+    inject(_Telemetry_ => {
+      service = _Telemetry_;
+    });
+  });
+
+  afterEach(function() {
+    clock.restore();
+  });
+
+  describe('Record', () => {
+    it('records a piece of telemetry', () => {
+      storageGetItem.withArgs('medic-greg-telemetry-db').returns('dbname');
+      storageGetItem
+        .withArgs('medic-greg-telemetry-date')
```

Comment:
I misread `greg-telemetry-date` as being the telemetry date in the Gregorian calendar :smile: